### PR TITLE
Add COMPILED State for ExecuteStatement af engine side

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

RUNNING -> COMPILED can be used to calculate to the duration of a statement being parsed, analyzed, and optimized

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationState.scala [hunk 1]
```diff
@@ -42,24 +52,14 @@ object OperationState extends Enumeration {
     case UNKNOWN => UKNOWN_STATE
   }
 
-  implicit def fromTOperationState(from: TOperationState): OperationState = from match {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationState.scala:45]: dead code

**Comment by yaooqinn** [comment]
cc @zhang1002 @ulysses-you @pan3793 @turboFei @yanghua

**Review by pan3793 — APPROVED** [review]
LGTM

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala [hunk 2]
```diff
@@ -93,6 +93,7 @@ class ExecuteStatement(
       // TODO: Make it configurable
       spark.sparkContext.addSparkListener(operationListener)
       result = spark.sql(statement)
+      // TODO( #921): COMPILED need consider eagerly executed commands
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:96]: cc @ulysses-you

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master for 1.3.0

**Pull request closed without merging**